### ISSUE TITLE
Other Interesting Api aquitecture

### ISSUE THREAD

**Issue by ctaity** [sumppen/WhatsApi4J#49]
https://github.com/hidespb/node-whatsapi

In this api, node proccesing is separated, i think this manner is more extensinble an clean.
What do you think? We can do the same in this api and add async callbacks with lambads ?

**Comment by sumppen** [comment]
I agree that the current architecture does need some improvement. It's mainly based on the php architecture, which is why certain things are done as they are.
Since you seem to have an implementation, maybe you could do a pull-request for it?

**Comment by ctaity** [comment]
i will do some tests with reacor http://projectreactor.io/ for async socket connection, and send a demo, it will take some time :D

**Comment by ctaity** [comment]
what do you think of this, i will start with create some interfaces and separate methods in categories.
Example final api:

wa.media().sendPhoto()
wa.message().send()
wa.contacts().sync()
wa.media().sendVideo()
wa.groups().requestGroupsList()
etc.....

the categories:

account, groups, media, message for now.

**Comment by ctaity** [comment]
yo can see first version of async socket in my fork. Any comments are apprcecietid
